src/timearray.jl

```diff
@@ -1,9 +1,9 @@
 ###### type definition ##########
 
-import Base: convert, copy, length, show, getindex, start, next, done, isempty,
-             endof, size, eachindex, ==, isequal, hash
+import Base: convert, copy, length, show, getindex, iterate,
+             lastindex, size, eachindex, ==, isequal, hash, ndims
 
-abstract type AbstractTimeSeries end
+abstract type AbstractTimeSeries{T,N,D} end
```

Comment:
A notable change: I make this abstract type being parametric, to help some iteration stuffs like `eltype`.